### ISSUE TITLE
Menu location incorrect and navigation disabled in safari after scrolling

### ISSUE THREAD

**Issue by muhlman** [Shopify/dawn#88]
**Describe the current behavior**
When scrolling the page and then scrolling back to the top, the menu sticks in an odd location (varies, not the same spot every time) and loses the ability to be clicked. 

**Describe the expected behavior**
The menu should return to the top of the page and remain clickable.

**Device information (browsers and operating system)**
- Safari 14.1.1
- macOS Version 11.4

**Possible solution**

**Additional context/screenshots**
I am using the theme on a fresh dev store with no plugins or theme customizations. 
<img width="2019" alt="Screen Shot 2021-06-30 at 7 14 44 PM" src="https://user-images.githubusercontent.com/11663981/124042724-78a1d580-d9d7-11eb-9c28-b7af9e5c9a52.png">

**Comment by chrisberthe** [comment]
Thanks @muhlman. I believe one of our team members have come across this once before cc @martinamarien.

**Comment by ShepelIgor** [comment]
Any problems in Chrome? or only in Safari?

**Comment by muhlman** [comment]
> Any problems in Chrome? or only in Safari?

No, I am only able to reproduce in Safari, works fine in Chrome, Edge, Brave and Firefox.

**Comment by henrybarn** [comment]
The `will-change` property can cause unpredictable behavior in Safari and appears to be the culprit here too.
